salt/modules/cmdmod.py

```diff
@@ -245,6 +245,10 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    # Remove whitespace from strings
+    if isinstance(cmd, six.string_types):
+        cmd = cmd.strip()
+
```

Comment:
I don't know why you felt the need to edit all that unrelated code in your latest commit.